contract/src/start-postalSvc.js

```diff
@@ -0,0 +1,106 @@
+/**
+ * @file core eval script* to start the postalSvc contract.
+ *
+ * * see test-gimix-proposal.js to make a script from this file.
+ *
+ * The `permit` export specifies the corresponding permit.
+ */
+// @ts-check
+
+import { E, Far } from '@endo/far';
+
+const { Fail } = assert;
+
+const trace = (...args) => console.log('start-postalSvc', ...args);
+
+const fail = msg => {
+  throw Error(msg);
+};
+
+/**
+ * ref https://github.com/Agoric/agoric-sdk/issues/8408#issuecomment-1741445458
+ *
+ * @param {ERef<import('@agoric/vats').NameAdmin>} namesByAddressAdmin
+ */
+const fixHub = async namesByAddressAdmin => {
+  /** @type {import('@agoric/vats').NameHub} */
+  const hub = Far('Hub work-around', {
+    lookup: async (addr, ...rest) => {
+      await E(namesByAddressAdmin).reserve(addr);
+      const addressAdmin = await E(namesByAddressAdmin).lookupAdmin(addr);
+      assert(addressAdmin, 'no admin???');
+      const addressHub = E(addressAdmin).readonly();
+      if (rest.length === 0) return addressHub;
+      await E(addressAdmin).reserve(rest[0]);
+      return E(addressHub).lookup(...rest);
+    },
+    has: _key => Fail`key space not well defined`,
+    entries: () => Fail`enumeration not supported`,
+    values: () => Fail`enumeration not supported`,
+    keys: () => Fail`enumeration not supported`,
+  });
+  return hub;
+};
+
+/**
+ * @param {BootstrapPowers} powers
+ * @param {{ options?: { postalSvc: {
+ *   bundleID: string;
+ * }}}} config
+ */
+export const startPostalSvc = async (powers, config) => {
+  const {
+    consume: { zoe, namesByAddressAdmin },
+    installation: {
+      // @ts-expect-error not statically known at genesis
+      produce: { postalSvc: produceInstallation },
+    },
+    instance: {
+      // @ts-expect-error not statically known at genesis
+      produce: { postalSvc: produceInstance },
+    },
+  } = powers;
+  const { bundleID = fail(`no bundleID; try test-gimix-proposal.js?`) } =
+    config.options?.postalSvc ?? {};
+
+  /** @type {Installation<import('./postalSvc').start>} */
+  const installation = await E(zoe).installBundleID(bundleID);
+  produceInstallation.resolve(installation);
+
+  const namesByAddress = await fixHub(namesByAddressAdmin);
+
+  const [IST, Invitation] = await Promise.all([
+    E(zoe).getFeeIssuer(),
+    E(zoe).getInvitationIssuer(),
+  ]);
+  const { instance } = await E(zoe).startInstance(
+    installation,
+    { IST, Invitation },
```

Comment:
> creatorFacet method to add issuers?
> governed API to add issuers?
> [#8547](https://github.com/Agoric/agoric-sdk/pull/8547#issue-2000141585)

It seems like this may be a common requirement. I wonder if it's possible for a contract to have a dynamic list of issuers from a nameHub. Could maybe be added to `StandardTerms` from zoe or specified in the `startInstance` function.

Reply:
I also wonder if [this](https://github.com/agoric-labs/dapp-swaparoo/blob/cb84362c35aeaf3bfb08df762f53efe6d8c3cc6e/contract/src/swaparoo.js#L137) is a better approach here, saving issuers on the fly. Tying postalSvc to a NameHub like `agoricNames` might preclude wider usage.

Reply:
Contracts usually add issuers that they are confident that they can rely on, so letting clients unilaterally add issuers is usually a no-no. But in this case, maybe it's ok, since we're just relaying assets that the parties choose to use?